File: BrainBoostFrontend/src/app/service/rest/user/user.service.ts
import { Injectable } from '@angular/core';
import {RestService} from '../rest.service';
import {LoginBean, RegisterBean, UserBean} from '../../../bean/user';
import {Store} from '@ngxs/store';
import {UserAction} from '../../../store/user/user.actions';

@Injectable({
  providedIn: 'root',
})
export class UserService {


  constructor(private readonly restService: RestService,
              private readonly store: Store) {

  }

  public login$(loginBean: LoginBean) {
    return this.restService.post$<UserBean>('/user/login', loginBean);
  }

  public register$(registerBean: RegisterBean) {
    return this.restService.post$<UserBean>('/user/register', registerBean);
  }

  public logout$() {
    return this.store.dispatch(new UserAction.Logout)
  }

}

File: src/main/java/dev/omel/brainboostbackend/service/UserService.java
package dev.omel.brainboostbackend.service;

import dev.omel.brainboostbackend.bean.UserBean;

public interface UserService {

    UserBean login(String username, String password) throws Exception;

    UserBean register(String username, String password, String firstName, String lastName) throws Exception;

    UserBean update(Long id, UserBean userBean) throws Exception;
}
